### PR TITLE
train_test_split_edges uses single random source

### PR DESCRIPTION
**Pull request by kessido**
current implementation of train_test_split_edges uses multiple randomness sources which harden the ability to reproduce experiment results. Given the name "torch_geometric" one would expect having set torch randomness state, that the produced split would be the same -> but this is not the case!

As such I suggest using only torch.randperm as the randomness source for the function.

Ido

### CODE REVIEW

**Comment by rusty1s** [comment]
I fully agree. The previous sampling technique has the advantage that it does not have to materialize `row.size(0)` entries, and hence is a little bit more memory-friendly.

**Comment by kessido** [comment]
Given that `row` itself already exist, materializing a vector of similar size seem reasonable, but it is your call.
Ido

**Comment by rusty1s** [comment]
That sounds good to me :) Thanks!
